src/shared/services/document-management/document-management.service.ts

```diff
@@ -26,4 +31,40 @@ export class DocumentManagementService {
         map(response => response.json())
       );
   }
+
+  createMediaViewer(caseField: CaseField): string {
+    let mediaViewer = {};
+    if (caseField.value) {
+      mediaViewer = {
+        document_binary_url: this.transformDocumentUrl(caseField.value.document_binary_url),
```

Comment:
`caseField.value` is duplicated a lot in this new code. Perhaps it'll be leaner to pass `caseField.value` directly to this `createMediaViewer` method rather than `caseField`. And invoke `createMediaViewer` only if `caseField.value` is not null so you don't have to check that in this service class multiple times

Reply:
Refactored by passing `caseField.value` as parameter